### PR TITLE
Added OptFieldSet func

### PR DESCRIPTION
**Pull request by yuce**
Cache type and cache size can only be specified through the OptFieldSet function, since they are only valid for set fields.

### CODE REVIEW

File: orm.go [hunk 1]
```diff
@@ -438,10 +435,17 @@ func (fo *FieldOptions) addOptions(options ...interface{}) error {
 // FieldOption is used to pass an option to index.Field function.
 type FieldOption func(options *FieldOptions) error
 
-// OptFieldCacheSize sets the cache size.
-func OptFieldCacheSize(size uint) FieldOption {
+// OptFieldInt adds a set field.
```
Comment (travisturner) [inline orm.go]: This comment is wrong.

File: orm.go [hunk 2]
```diff
@@ -438,10 +435,17 @@ func (fo *FieldOptions) addOptions(options ...interface{}) error {
...
+// Specify CacheTypeDefault for the default cache type.
+// Specify 0 for the default cache size.
```
Comment (travisturner) [inline orm.go]: It seems weird to use the default cache size when the value provided is 0. At that point, why not provide a `CacheSizeDefault` const for consistency with the `CacheTypeDefault ` option? I think if someone sets `cacheSize` to 0 it should effectively act like `CacheTypeNone`.

Comment (yuce) [reply orm.go]: Added `CacheSizeDefault` const.

File: orm.go (hunk 1 above)
Comment (yuce) [reply orm.go]: Fixed.

**Comment by yuce** [comment]
@jaffee Right, if cache size is <= 0, i it is not sent to the server side, so the cache size is set to the default on the server side. Line 392 in `orm.go`:
```go
		if fo.cacheSize > 0 {
			mopt["cacheSize"] = fo.cacheSize
		}
```

**Comment by yuce** [comment]
great!

**Pull request merged**